### PR TITLE
adding cad-to-dagmc

### PR DESCRIPTION
**Pull request by shimwell**
This PR adds a recipe for cad-to-dagmc. This makes use of an existing conda forge packages (moab, ocp and cadquery) to make files for another conda forge package (dagmc).

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/cad_to_dagmc/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+build:
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+  skip: true  # [win]
+  skip: True  # [py<310]
+  skip: True  # [py>=312]
+
+requirements:
+  host:
+    - python >=3.10,<3.12
```
Comment (shimwell) [inline recipes/cad_to_dagmc/meta.yaml:21]:
```suggestion
    - python
```

File: recipes/cad_to_dagmc/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    - pip
+  run:
+    - python >=3.10,<3.12
```
Comment (shimwell) [inline recipes/cad_to_dagmc/meta.yaml:24]:
```suggestion
    - python
```

**Comment by shimwell** [comment]
@conda-forge/help-python ready for review. Many thanks

File: recipes/cad_to_dagmc/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: b6b7f34dec0d54a19a9476a2c5c157ef23eb3d6796af86dee6798cdc2420fda9
+
+build:
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+  skip: True  # [win]
+  skip: True  # [py<310]
+  skip: True  # [py>=312]
```
Comment (xhochy) [inline recipes/cad_to_dagmc/meta.yaml:17]: This looks like it could be noarch. Why aren't you using that?
```suggestion
  noarch: python
```

Comment (shimwell) [reply recipes/cad_to_dagmc/meta.yaml:17]: Thanks for this. I thought all the dependencies had to be noarch for the recipe to also be noarch. So I noticed [moab](https://github.com/conda-forge/moab-feedstock/blob/main/recipe/meta.yaml) and other dependencies are not noarch and thought I therefore couldn't be.

Glad to hear that I'm wrong there. Keen to make this a noarch recipe.

File: recipes/cad_to_dagmc/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+  noarch: python
+
+requirements:
+  host:
+    - python
+    - pip
+  run:
+    - python
```
Comment (shimwell) [inline recipes/cad_to_dagmc/meta.yaml:22]:
```suggestion
    - python >=3.10
```

File: recipes/cad_to_dagmc/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  sha256: b6b7f34dec0d54a19a9476a2c5c157ef23eb3d6796af86dee6798cdc2420fda9
+
+build:
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+  noarch: python
+
+requirements:
+  host:
+    - python
```
Comment (shimwell) [inline recipes/cad_to_dagmc/meta.yaml:19]:
```suggestion
    - python >=3.10
```

File: recipes/cad_to_dagmc/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: b6b7f34dec0d54a19a9476a2c5c157ef23eb3d6796af86dee6798cdc2420fda9
+
+build:
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+  noarch: python
```
Comment (shimwell) [inline recipes/cad_to_dagmc/meta.yaml:15]: I notice the Ci is now failing the windows test so perhaps best to put that skip windows back in

```suggestion
  skip: True  # [win]
  noarch: python
```

File: recipes/cad_to_dagmc/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: b6b7f34dec0d54a19a9476a2c5c157ef23eb3d6796af86dee6798cdc2420fda9
+
+build:
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+  skip: True  # [win]
```
Comment (shimwell) [inline recipes/cad_to_dagmc/meta.yaml:15]:
```suggestion
```

File: recipes/cad_to_dagmc/meta.yaml (hunk 3 above)
Comment (shimwell) [reply recipes/cad_to_dagmc/meta.yaml:17]: I had a go at moving to noarch but could no longer skip the windows build.

Unfortunately windows builds won't work for now as moab doesn't build a conda package for windows [link to moab meta.yml skip windows line](https://github.com/conda-forge/moab-feedstock/blob/7707e337d01852ee60cfc8baed87501e15ff7ff5/recipe/meta.yaml#L48)

I believe there is a longer term plan to move the moab conda build from autotools to cmake then add windows support. So I shall keep an eye on that and then could come back and have another go at making this package noarch in the future if/when moab builds conda packages for windows.

Perhaps noarch is not necessary for now?

**Comment by xhochy** [comment]
If windows fails because of a dependency issue, this is OK. We are building noarch packages on Linux. It simply means people will not be able to install the package on Windows as long as the dependency doesn't exist.

**Comment by shimwell** [comment]
> If windows fails because of a dependency issue, this is OK. We are building noarch packages on Linux. It simply means people will not be able to install the package on Windows as long as the dependency doesn't exist.

Thanks for letting me know. I shall go back to making this a noarch.

**Review by xhochy — APPROVED** [review]

**Pull request merged**